sdk/eventgrid/eventgrid/samples-dev/consumeEventsFromServiceBusQueue.ts

```diff
@@ -63,9 +63,9 @@ async function main() {
   // Stop processing events and exit.
   await closer.close();
   await receiver.close();
-  process.exit();
 }
 
 main().catch((err) => {
   console.error("The sample encountered an error:", err);
+  process.exit(1);
```

Comment:
Good question!  This actually doesn't matter to the test runner as it only invokes the sample's `main()` function.  I was just following the convention of most existing samples.

Reply:
It's possible that `process.exit(1)` was used at some point? @witemple-msft I wonder whether you have any ideas as you worked with dev-tool and samples before?

Reply:
This is just a conventional pattern for async scripts in JavaScript. Instead of rethrowing the error, you also exit(1). The problem, fundamentally, is that an unhandled Promise rejection does not result in an error status in some (all?) versions of Node, so explicitly calling `exit` is the only way for the script itself to report that it failed to the shell.